File: src/main/java/me/celestialfault/celestialwynn/config/FOVScaling.java
package me.celestialfault.celestialwynn.config;

import net.minecraft.client.network.AbstractClientPlayerEntity;

public enum FOVScaling {
	VANILLA,
	ARCTANGENT,
	SPRINT_ONLY;

	public float calculate(float fovMultiplier, AbstractClientPlayerEntity player) {
		return switch(this) {
			case ARCTANGENT -> 1f + (float) (Math.atan(2d * Math.PI * (fovMultiplier - 1d)) / (2d * Math.PI));
			case SPRINT_ONLY -> 1f + (player.isSprinting() ? 0.15f : 0);
			default -> fovMultiplier;
		};
	}
}

File: src/main/kotlin/dev/celestialfault/celestialwynn/enums/FOVScaling.kt
package dev.celestialfault.celestialwynn.enums

import net.minecraft.client.network.AbstractClientPlayerEntity
import kotlin.math.atan

enum class FOVScaling {
	VANILLA,
	ARCTANGENT,
	SPRINT_ONLY;

	fun calculate(fovMultiplier: Float, player: AbstractClientPlayerEntity): Float = when(this) {
		ARCTANGENT -> 1f + (atan(2.0 * Math.PI * (fovMultiplier - 1.0)) / (2.0 * Math.PI)).toFloat()
		SPRINT_ONLY -> 1f + (if(player.isSprinting) 0.15f else 0f)
		VANILLA -> fovMultiplier
	}
}
